### PR TITLE
Make PagedListResponse iterable

### LINKED ISSUE #153: Make PagedListResponse Iterable
We should make `PagedListResponse` iterable. This would clean up the code:

```php
$devices = $client->getDevices($parent, $registry);

// current code, not intuitive
foreach ($devices->iterateAllElements() as $device) {
    printf('Device ID: %s' . PHP_EOL, $device->getId());
}

// better code
foreach ($devices as $device) {
    printf('Device ID: %s' . PHP_EOL, $device->getId());
}
```

Another issue is the second one does not throw an error, it just fails to print out anything, which is the same behavior as if it had no elements. So it's very confusing!

### PR DESCRIPTION
**Pull request by michaelbausor**
PTAL, and let me know whether we need more or clearer documentation. cc @sirtorry, @tmatsuo 

Closes #153

### CODE REVIEW

**Comment by michaelbausor** [comment]
@bshaffer added examples in the class comments - WDYT?

File: src/PagedListResponse.php [hunk 1]
```diff
@@ -43,12 +44,42 @@
...
+ *  - As a single iterable used in a foreach loop or via the getIterator method
  *  - As pages of elements, using the getPage and iteratePages methods
  *  - As fixed size collections of elements, using the
  *    getFixedSizeCollection and iterateFixedSizeCollections methods
+ *
+ * Example of using PagedListResponse as an iterator:
+ * ```
+ * $pagedListResponse = $client->getList(...);
+ * foreach ($pagedListResponse as $element) {
+ *   // doSomethingWith($element);
```
Comment (dwsupplee) [inline src/PagedListResponse.php]: Could we please use 4 spaces?

Comment (bshaffer) [reply src/PagedListResponse.php]: agreed, these are great but let's use 4 spaces!!

Comment (michaelbausor) [reply src/PagedListResponse.php]: Done!

**Comment by bshaffer** [comment]
What a guy!

**Pull request merged**